File: base/src/main/java/com/lotteryviewer/base/app/SimpleActivityLifecycleCallbacks.java
package com.lotteryviewer.base.app;

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

import androidx.annotation.NonNull;

/**
 * @Author: duke
 * @DateTime: 2021-03-17 11:13:45
 * @Description: 功能描述：
 */
public class SimpleActivityLifecycleCallbacks implements Application.ActivityLifecycleCallbacks {

    /**
     * Called as the first step of the Activity being created. This is always called before
     * {@link Activity#onCreate}.
     */
    @Override
    public void onActivityPreCreated(@NonNull Activity activity, @androidx.annotation.Nullable Bundle savedInstanceState) {

    }

    /**
     * Called when the Activity calls {@link Activity#onCreate super.onCreate()}.
     */
    @Override
    public void onActivityCreated(@NonNull Activity activity, @androidx.annotation.Nullable Bundle savedInstanceState) {

    }

    /**
     * Called as the last step of the Activity being created. This is always called after
     * {@link Activity#onCreate}.
     */
    @Override
    public void onActivityPostCreated(@NonNull Activity activity, @androidx.annotation.Nullable Bundle savedInstanceState) {

    }

    /**
     * Called as the first step of the Activity being started. This is always called before
     * {@link Activity#onStart}.
     */
    @Override
    public void onActivityPreStarted(@NonNull Activity activity) {

    }

    /**
     * Called when the Activity calls {@link Activity#onStart super.onStart()}.
     */
    @Override
    public void onActivityStarted(@NonNull Activity activity) {

    }

    /**
     * Called as the last step of the Activity being started. This is always called after
     * {@link Activity#onStart}.
     */
    @Override
    public void onActivityPostStarted(@NonNull Activity activity) {

    }

    /**
     * Called as the first step of the Activity being resumed. This is always called before
     * {@link Activity#onResume}.
     */
    @Override
    public void onActivityPreResumed(@NonNull Activity activity) {

    }

    /**
     * Called when the Activity calls {@link Activity#onResume super.onResume()}.
     */
    @Override
    public void onActivityResumed(@NonNull Activity activity) {

    }

    /**
     * Called as the last step of the Activity being resumed. This is always called after
     * {@link Activity#onResume} and {@link Activity#onPostResume}.
     */
    @Override
    public void onActivityPostResumed(@NonNull Activity activity) {

    }

    /**
     * Called as the first step of the Activity being paused. This is always called before
     * {@link Activity#onPause}.
     */
    @Override
    public void onActivityPrePaused(@NonNull Activity activity) {

    }

    /**
     * Called when the Activity calls {@link Activity#onPause super.onPause()}.
     */
    @Override
    public void onActivityPaused(@NonNull Activity activity) {

    }

    /**
     * Called as the last step of the Activity being paused. This is always called after
     * {@link Activity#onPause}.
     */
    @Override
    public void onActivityPostPaused(@NonNull Activity activity) {

    }

    /**
     * Called as the first step of the Activity being stopped. This is always called before
     * {@link Activity#onStop}.
     */
    @Override
    public void onActivityPreStopped(@NonNull Activity activity) {

    }

    /**
     * Called when the Activity calls {@link Activity#onStop super.onStop()}.
     */
    @Override
    public void onActivityStopped(@NonNull Activity activity) {

    }

    /**
     * Called as the last step of the Activity being stopped. This is always called after
     * {@link Activity#onStop}.
     */
    @Override
    public void onActivityPostStopped(@NonNull Activity activity) {

    }

    /**
     * Called as the first step of the Activity saving its instance state. This is always
     * called before {@link Activity#onSaveInstanceState}.
     */
    @Override
    public void onActivityPreSaveInstanceState(@NonNull Activity activity, @NonNull Bundle outState) {

    }

    /**
     * Called when the Activity calls
     * {@link Activity#onSaveInstanceState super.onSaveInstanceState()}.
     */
    @Override
    public void onActivitySaveInstanceState(@NonNull Activity activity, @NonNull Bundle outState) {

    }

    /**
     * Called as the last step of the Activity saving its instance state. This is always
     * called after{@link Activity#onSaveInstanceState}.
     */
    @Override
    public void onActivityPostSaveInstanceState(@NonNull Activity activity, @NonNull Bundle outState) {

    }

    /**
     * Called as the first step of the Activity being destroyed. This is always called before
     * {@link Activity#onDestroy}.
     */
    @Override
    public void onActivityPreDestroyed(@NonNull Activity activity) {

    }

    /**
     * Called when the Activity calls {@link Activity#onDestroy super.onDestroy()}.
     */
    @Override
    public void onActivityDestroyed(@NonNull Activity activity) {

    }

    /**
     * Called as the last step of the Activity being destroyed. This is always called after
     * {@link Activity#onDestroy}.
     */
    @Override
    public void onActivityPostDestroyed(@NonNull Activity activity) {

    }

}

File: base/src/main/java/com/lotteryviewer/base/app/SimpleActivityLifecycleCallbacks.kt
package com.lotteryviewer.base.app

import android.app.Activity
import android.app.Application.ActivityLifecycleCallbacks
import android.os.Bundle

/**
 * @Author: duke
 * @DateTime: 2021-03-17 11:13:45
 * @Description: 功能描述：
 */
open class SimpleActivityLifecycleCallbacks : ActivityLifecycleCallbacks {
    /**
     * Called as the first step of the Activity being created. This is always called before
     * [Activity.onCreate].
     */
    override fun onActivityPreCreated(activity: Activity, savedInstanceState: Bundle?) {}

    /**
     * Called when the Activity calls [super.onCreate()][Activity.onCreate].
     */
    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {}

    /**
     * Called as the last step of the Activity being created. This is always called after
     * [Activity.onCreate].
     */
    override fun onActivityPostCreated(activity: Activity, savedInstanceState: Bundle?) {}

    /**
     * Called as the first step of the Activity being started. This is always called before
     * [Activity.onStart].
     */
    override fun onActivityPreStarted(activity: Activity) {}

    /**
     * Called when the Activity calls [super.onStart()][Activity.onStart].
     */
    override fun onActivityStarted(activity: Activity) {}

    /**
     * Called as the last step of the Activity being started. This is always called after
     * [Activity.onStart].
     */
    override fun onActivityPostStarted(activity: Activity) {}

    /**
     * Called as the first step of the Activity being resumed. This is always called before
     * [Activity.onResume].
     */
    override fun onActivityPreResumed(activity: Activity) {}

    /**
     * Called when the Activity calls [super.onResume()][Activity.onResume].
     */
    override fun onActivityResumed(activity: Activity) {}

    /**
     * Called as the last step of the Activity being resumed. This is always called after
     * [Activity.onResume] and [Activity.onPostResume].
     */
    override fun onActivityPostResumed(activity: Activity) {}

    /**
     * Called as the first step of the Activity being paused. This is always called before
     * [Activity.onPause].
     */
    override fun onActivityPrePaused(activity: Activity) {}

    /**
     * Called when the Activity calls [super.onPause()][Activity.onPause].
     */
    override fun onActivityPaused(activity: Activity) {}

    /**
     * Called as the last step of the Activity being paused. This is always called after
     * [Activity.onPause].
     */
    override fun onActivityPostPaused(activity: Activity) {}

    /**
     * Called as the first step of the Activity being stopped. This is always called before
     * [Activity.onStop].
     */
    override fun onActivityPreStopped(activity: Activity) {}

    /**
     * Called when the Activity calls [super.onStop()][Activity.onStop].
     */
    override fun onActivityStopped(activity: Activity) {}

    /**
     * Called as the last step of the Activity being stopped. This is always called after
     * [Activity.onStop].
     */
    override fun onActivityPostStopped(activity: Activity) {}

    /**
     * Called as the first step of the Activity saving its instance state. This is always
     * called before [Activity.onSaveInstanceState].
     */
    override fun onActivityPreSaveInstanceState(activity: Activity, outState: Bundle) {}

    /**
     * Called when the Activity calls
     * [super.onSaveInstanceState()][Activity.onSaveInstanceState].
     */
    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}

    /**
     * Called as the last step of the Activity saving its instance state. This is always
     * called after[Activity.onSaveInstanceState].
     */
    override fun onActivityPostSaveInstanceState(activity: Activity, outState: Bundle) {}

    /**
     * Called as the first step of the Activity being destroyed. This is always called before
     * [Activity.onDestroy].
     */
    override fun onActivityPreDestroyed(activity: Activity) {}

    /**
     * Called when the Activity calls [super.onDestroy()][Activity.onDestroy].
     */
    override fun onActivityDestroyed(activity: Activity) {}

    /**
     * Called as the last step of the Activity being destroyed. This is always called after
     * [Activity.onDestroy].
     */
    override fun onActivityPostDestroyed(activity: Activity) {}
}